### PR TITLE
Issue/13 tooltip

### PR DESCRIPTION
**Pull request by Wh0Kn0w5**
A draft PR was requested, and the most recent commit notes show my current progress.

Have the ContextChanged method available, removed the "Loaded" value from the base xaml and replaced with the DataContextChanged. Currently getting some ORE errors on my current progress with the .cs, would like some more pointers of what and where to add.

### CODE REVIEW

**Comment by pileofwolves** [comment]
Marked the PR as a draft.

The behavior currently appears to be equivalent to previous logic: when a valid DataContext gets assigned, prepare a tooltip. It doesn't reset the tooltip if the DataContext changes again, though - I think that's where your problem lies. Note this snippet in `CreateTooltipContent()`:

```csharp
// Ensure we're not duplicating the tooltip contents
if (parts.Count != 0)
    return;
```

This prevents the tooltip from refreshing when the item box gets a new item assigned and its DataContext changes. The Inlines collection should be cleared, so the above check does not block the refresh, and perhaps the check can also be removed entirely at that point (though not 100% sure about that, by all means test).

**Comment by Wh0Kn0w5** [comment]
Ok, after applying the changes suggested, it seems that we're working now.

Do you need the pull request set as "Ready for review" before you can see if it works with what you have?

File: Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@
 using Finmer.Utility;
 using System;
 using System.Windows;
+using System.ComponentModel;
+using System.Diagnostics;
```
Comment (pileofwolves) [inline Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs:16]: These new `using` directives are unused; please remove them.

File: Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs [hunk 2]
```diff
@@ -26,23 +28,23 @@ namespace Finmer.Views
     public partial class ItemTooltipView
     {
 
+        private WeakReference<ItemTooltipView> m_CurrentTip;
```
Comment (pileofwolves) [inline Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs:31]: This field is unused; please remove it

File: Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs [hunk 3]
```diff
@@ -216,6 +218,10 @@ private static UIElement CreateCoinImage()
             return container;
         }
 
+        private void UserControl_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
```
Comment (pileofwolves) [inline Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs:221]: This method is unused; please remove it

**Review by pileofwolves — CHANGES_REQUESTED** [review]
The 'draft' status doesn't prevent me (or anyone else for that matter) from looking at or testing the changes, it just visually marks the PR as not being ready to merge.

The change looks like it'll do the trick, just some cleanup to do.

File: Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs (hunk 3 above)
Comment (Wh0Kn0w5) [reply Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs:221]: not gonna lie, didn't even know this part of the code existed...

**Comment by Wh0Kn0w5** [comment]
changes pushed, should show on your end shortly.

**Review by pileofwolves — APPROVED** [review]

**Comment by pileofwolves** [comment]
Yeah, they show up as soon as you push the branch on your fork.

Thanks for the PR, I'll merge it when I get home in a bit.

File: Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs (hunk 3 above)
Comment (pileofwolves) [reply Finmer.Game/Views/Parts/ItemTooltipView.xaml.cs]: > 
> 
> not gonna lie, didn't even know this part of the code existed...

Be sure to diff your changes (either `git diff` or right-click > 'Compare with Unmodified' in Visual Studio). A quick check over your changes before committing / pushing can help you with finding obvious problems and making sure the change looks sane.

**Pull request merged**